frontend/src/components/ui/blockGenerator/Folder-uploadBlock.jsx

```diff
@@ -1,55 +1,61 @@
 import { useEffect, useRef, useState } from "react";
-import { trimQuotes } from "@/utils/blockUtils";
 
 export const FolderBlock = ({ blockId, block, setFocusAction, history }) => {
   const fileInput = useRef();
+  const uniqueKey = `folderName_${blockId}`;
   const [renderPath, setRenderPath] = useState(null);
+  const [folderName, setFolderName] = useState(
+    () => localStorage.getItem(uniqueKey) || null,
+  );
+  const [filePaths, setFilePaths] = useState(() => {
+    const storedPaths = localStorage.getItem(`filePaths_${blockId}`);
+    return storedPaths ? JSON.parse(storedPaths) : [];
+  });
 
   useEffect(() => {
-    const type = block?.action?.parameters["path"]?.type;
-    if (history && type !== "fileLoad") {
-      let fileName = block?.action?.parameters["path"]?.value;
-      const fileSplit = fileName.split("/");
-      if (fileSplit.length > 1) {
-        fileName = fileSplit.at(-1);
-      }
-      fileName = trimQuotes(fileName);
-      const s3Url = history + "/" + fileName;
+    if (history && filePaths.length > 0 && folderName) {
+      const formattedValue = `[${filePaths.map((file) => `"${file}"`).join(", ")}]`;
       setFocusAction((draft) => {
-        draft.data[blockId].action.parameters["path"].value = s3Url;
-        draft.data[blockId].action.parameters["path"].type = "blob";
+        draft.data[blockId].action.parameters["path"].type = "folder";
+        draft.data[blockId].action.parameters["path"].value = formattedValue;
       });
-      setRenderPath(fileName);
+      setRenderPath(folderName);
     }
-  }, [block]);
+  }, [block, folderName, filePaths]);
 
   const processFiles = (files) => {
     const filePaths = [];
     const readDirectory = (dir, path = "") => {
       for (let i = 0; i < dir.length; i++) {
         const file = dir[i];
-        const fullPath = path ? `${path}/${file.name}` : file.name;
+        const fullPath = path ? `${path}/${file.path}` : file.path;
         if (file.webkitDirectory) {
           readDirectory(file.children, fullPath);
         } else {
           filePaths.push(fullPath);
         }
       }
     };
-
     readDirectory(files, "");
     return filePaths;
   };
 
   const loadFiles = (e) => {
     const files = Array.from(fileInput.current.files);
     const filePaths = processFiles(files);
-    const value = filePaths.join(", ");
+    const formattedValue = `[${filePaths.map((file) => `"${file}"`).join(", ")}]`;
+    const firstFilePath = filePaths[0];
+    const pathSegments = firstFilePath.split(/[/\\]/);
+    const extractedFolderName = pathSegments[pathSegments.length - 2];
+    setFolderName(extractedFolderName);
+    setFilePaths(filePaths);
+    localStorage.setItem(uniqueKey, extractedFolderName);
```

Comment:
Why do we set localStorage?

Reply:
I set `localStorage `to retain the folder name and file paths across page reloads or runs. If I do not use it like this way, when I drag and drop the 1st folder-upload block it works fine. if I drag and drop the 2nd folder-upload block in the same pipeline, it does not get a name, also does not upload any file.